### PR TITLE
[Docs] Metrics

### PR DESCRIPTION
**Pull request by xinke-wang**

### CODE REVIEW

File: docs/zh_cn/basic_concepts/evaluation.md [hunk 1]
```diff
@@ -1,3 +1,194 @@
...
+
+MMOCR 基于 {external+mmengine:doc}`MMEngine: BaseMetric <design/evaluation>` 基类实现了常用的文本检测、文本识别以及关键信息抽取任务的评测指标，用户可以通过修改配置文件中的 `val_evaluator` 与 `test_evaluator` 字段来便捷地指定验证与测试阶段采用的评测方法。例如，以下配置展示了如何在文本检测算法中使用 `HmeanIOUMetric` 来评测模型性能。
+
+```python
+val_evaluator = dict(type='HmeanIOUMetric')
+test_evaluator = val_evaluator
+```
+
+```{tip}
+更多评测相关配置请参考评[测配置教程](../user_guides/config.md#评测配置)。
```
Comment (gaotongxiao) [inline docs/zh_cn/basic_concepts/evaluation.md:13]:
```suggestion
更多评测相关配置请参考[评测配置教程](../user_guides/config.md#评测配置)。
```

File: docs/en/basic_concepts/evaluation.md [hunk 2]
```diff
@@ -1,3 +1,193 @@
...
+```
+
+As shown in the following table, MMOCR currently supports 5 evaluation metrics for text detection, text recognition, and key information extraction tasks, including `HmeanIOUMetric`, `WordMetric`, `CharMetric`, `OneMinusNEDMetric`, and `F1Metric`.
+
+|                                                                             |         |                                                   |                                                                       |
+| --------------------------------------------------------------------------- | ------- | ------------------------------------------------- | --------------------------------------------------------------------- |
+| Metric                                                                      | Task    | Input Field                                       | Output Field                                                          |
+| [HmeanIOUMetric](mmocr.evaluation.metrics.hmean_iou_metric.HmeanIOUMetric)  | TextDet | `pred_polygons`<br>`pred_scores`<br>`gt_polygons` | `recall`<br>`precision`<br>`hmean`                                    |
```
Comment (gaotongxiao) [inline docs/en/basic_concepts/evaluation.md:21]: invalid links

File: docs/en/basic_concepts/evaluation.md [hunk 3]
```diff
@@ -1,3 +1,193 @@
...
+
+```{math}
+F_1=(1+\beta^2)\cdot\frac{PR}{\beta^2\cdot P+R} = \frac{2PR}{P+R}
+```
+
+In MMOCR, the calculation of `HmeanIOUMetric` can be summarized as the following steps:
+
+1. Filter out invalid predictions
+
+   - Filter out predictions with a score lower than `pred_score_thrs`
```
Comment (gaotongxiao) [inline docs/en/basic_concepts/evaluation.md:47]:
```suggestion
   - Filter out predictions with a score is lower than `pred_score_thrs`
```

File: docs/en/basic_concepts/evaluation.md [hunk 4]
```diff
@@ -1,3 +1,193 @@
...
+
+### CharMetric
+
+`CharMetric` implements **character-level** text recognition evaluation metrics that are **case-insensitive**.
+
+```python
+# Use CharMetric for text recognition task
+val_evaluator = [dict(type='CharMetric')]
+
+# In addition, MMOCR also supports the combination evaluation of multiple metrics for the same task, such as using WordMetric and CharMetric at the same time
```
Comment (gaotongxiao) [inline docs/en/basic_concepts/evaluation.md:114]:
```suggestion
# In addition, MMOCR also supports the combined evaluation of multiple metrics for the same task, such as using WordMetric and CharMetric at the same time
```

**Review by gaotongxiao — APPROVED** [review]

File: docs/zh_cn/basic_concepts/evaluation.md [hunk 5]
```diff
@@ -1,3 +1,194 @@
...
+   值得注意的是，`pred_score_thrs` 默认将**自动搜索**一定范围内的**最佳阈值**，用户也可以通过手动修改配置文件来自定义搜索范围：
+
+   ```python
+   # HmeanIOUMetric 默认以 0.1 为步长搜索 [0.3, 0.9] 范围内的最佳得分阈值
+   val_evaluator = dict(type='HmeanIOUMetric', pred_score_thrs=dict(start=0.3, stop=0.9, step=0.1))
+   ```
+
+2. 计算 IoU 矩阵
+
+   - 在数据处理阶段，`HmeanIOUMetric` 会计算并维护一个 {math}`M \times N` 的 IoU 矩阵 `iou_metric`，以方便后续的边界盒配对步骤。其中，M 和 N 分别为标签边界盒与预测边界盒的数量。由此，该矩阵的每个元素都存放了第 m 个标签边界盒与第 n 个预测边界盒之间的交并比（IoU）。
```
Comment (Harold-lkk) [inline docs/zh_cn/basic_concepts/evaluation.md:59]: N is the filtered predictions

File: docs/zh_cn/basic_concepts/evaluation.md [hunk 6]
```diff
@@ -1,3 +1,194 @@
...
+
+     针对现有匹配机制中的不完善之处，MMOCR 算法库实现了一套更高效的匹配策略，用以最大化匹配数目。
+
+     ```python
+     # 指定采用 'max_matching' 匹配策略
+     val_evaluator = dict(type='HmeanIOUMetric', strategy='max_matching')
+     ```
+
+   ```{note}
+   我们建议面向学术研究的开发用户采用默认的 `vanilla` 匹配策略，以保证与其他论文的对比结果保持一致。而面向工业应用的开发用户则可以采用 `max_matching` 匹配策略，以获得更高的性能。
```
Comment (Harold-lkk) [inline docs/zh_cn/basic_concepts/evaluation.md:84]: 采用 `max_matching` 匹配策略，以获得更准确/精准/实际/真实的结果。

File: docs/en/basic_concepts/evaluation.md [hunk 7]
```diff
@@ -1,193 +1,3 @@
...
-    def compute_metrics(self, results: List):
-    """ compute_metric receives the results of the process method as input and returns the evaluation results.
-    """
-        pass
-```
-
-```{note}
-More details can be found in {external+mmengine:doc}`MMEngine Documentation: BaseMetric <design/evaluation>`.
-```
+Coming Soon!
```
Comment (gaotongxiao) [inline docs/en/basic_concepts/evaluation.md:3]: Put the tutorials back to Evaluation in order to be consistent with OpenMMLab 2.0 conventions.

**Review by gaotongxiao — CHANGES_REQUESTED** [review]

**Review by gaotongxiao — APPROVED** [review]

**Pull request merged**